libsolidity/formal/SMTEncoder.cpp

```diff
@@ -883,6 +886,102 @@ void SMTEncoder::endVisit(ElementaryTypeNameExpression const& _typeName)
 	m_context.createExpression(_typeName, result.second);
 }
 
+namespace
+{
+unsigned dimensionCount(TypePointer _type)
+{
+	unsigned dimensionCount = 0;
+	while (_type->category() == Type::Category::Array || _type->category() == Type::Category::Mapping)
+	{
+		++dimensionCount;
+		if (_type->category() == Type::Category::Array)
+			_type = dynamic_cast<ArrayType const *>(_type)->baseType();
+		else if (_type->category() == Type::Category::Mapping)
+			_type = dynamic_cast<MappingType const *>(_type)->valueType();
+		else
```

Comment:
We can remove the else, here it's pretty clear it won't happen

Reply:
Right now that's true. But in case something changes in the future, I would keep it so it's clear. And it will not accidentally end up in an infinite loop if somebody changes the condition but forget to update the body.